--- bin/linux-ppc-64/sonar.sh
+++ bin/linux-ppc-64/sonar.sh
@@ -371,6 +371,7 @@
         else
             su - $RUN_AS_USER -c "\"$REALPATH\" $2"
         fi
+        RETVAL=$?
 
         # Now that we are the original user again, we may need to clean up the lock file.
         if [ "X$LOCKPROP" != "X" ]
@@ -386,7 +387,7 @@
             fi
         fi
 
-        exit 0
+        exit $RETVAL
     fi
 }
 
--- bin/linux-x86-32/sonar.sh
+++ bin/linux-x86-32/sonar.sh
@@ -371,6 +371,7 @@
         else
             su - $RUN_AS_USER -c "\"$REALPATH\" $2"
         fi
+        RETVAL=$?
 
         # Now that we are the original user again, we may need to clean up the lock file.
         if [ "X$LOCKPROP" != "X" ]
@@ -386,7 +387,7 @@
             fi
         fi
 
-        exit 0
+        exit $RETVAL
     fi
 }
 
--- bin/linux-x86-64/sonar.sh
+++ bin/linux-x86-64/sonar.sh
@@ -371,6 +371,7 @@
         else
             su - $RUN_AS_USER -c "\"$REALPATH\" $2"
         fi
+        RETVAL=$?
 
         # Now that we are the original user again, we may need to clean up the lock file.
         if [ "X$LOCKPROP" != "X" ]
@@ -386,7 +387,7 @@
             fi
         fi
 
-        exit 0
+        exit $RETVAL
     fi
 }
 
--- bin/macosx-universal-32/sonar.sh
+++ bin/macosx-universal-32/sonar.sh
@@ -371,6 +371,7 @@
         else
             su - $RUN_AS_USER -c "\"$REALPATH\" $2"
         fi
+        RETVAL=$?
 
         # Now that we are the original user again, we may need to clean up the lock file.
         if [ "X$LOCKPROP" != "X" ]
@@ -386,7 +387,7 @@
             fi
         fi
 
-        exit 0
+        exit $RETVAL
     fi
 }
 
--- bin/macosx-universal-64/sonar.sh
+++ bin/macosx-universal-64/sonar.sh
@@ -371,6 +371,7 @@
         else
             su - $RUN_AS_USER -c "\"$REALPATH\" $2"
         fi
+        RETVAL=$?
 
         # Now that we are the original user again, we may need to clean up the lock file.
         if [ "X$LOCKPROP" != "X" ]
@@ -386,7 +387,7 @@
             fi
         fi
 
-        exit 0
+        exit $RETVAL
     fi
 }
 
--- bin/solaris-sparc-32/sonar.sh
+++ bin/solaris-sparc-32/sonar.sh
@@ -371,6 +371,7 @@
         else
             su - $RUN_AS_USER -c "\"$REALPATH\" $2"
         fi
+        RETVAL=$?
 
         # Now that we are the original user again, we may need to clean up the lock file.
         if [ "X$LOCKPROP" != "X" ]
@@ -386,7 +387,7 @@
             fi
         fi
 
-        exit 0
+        exit $RETVAL
     fi
 }
 
--- bin/solaris-sparc-64/sonar.sh
+++ bin/solaris-sparc-64/sonar.sh
@@ -371,6 +371,7 @@
         else
             su - $RUN_AS_USER -c "\"$REALPATH\" $2"
         fi
+        RETVAL=$?
 
         # Now that we are the original user again, we may need to clean up the lock file.
         if [ "X$LOCKPROP" != "X" ]
@@ -386,7 +387,7 @@
             fi
         fi
 
-        exit 0
+        exit $RETVAL
     fi
 }
 
--- bin/solaris-x86-32/sonar.sh
+++ bin/solaris-x86-32/sonar.sh
@@ -371,6 +371,7 @@
         else
             su - $RUN_AS_USER -c "\"$REALPATH\" $2"
         fi
+        RETVAL=$?
 
         # Now that we are the original user again, we may need to clean up the lock file.
         if [ "X$LOCKPROP" != "X" ]
@@ -386,7 +387,7 @@
             fi
         fi
 
-        exit 0
+        exit $RETVAL
     fi
 }
 
